packages/arb-token-bridge-ui/src/token-bridge-sdk/EthTeleportStarter.ts

```diff
@@ -0,0 +1,123 @@
+import { EthL1L3Bridger, getL2Network } from '@arbitrum/sdk'
+import { BigNumber, constants } from 'ethers'
+import {
+  ApproveNativeCurrencyProps,
+  BridgeTransferStarter,
+  BridgeTransferStarterProps,
+  RequiresNativeCurrencyApprovalProps,
+  TransferEstimateGas,
+  TransferProps,
+  TransferType
+} from './BridgeTransferStarter'
+import { getL2ConfigForTeleport } from './teleport'
+import { getAddressFromSigner } from './utils'
+
+export class EthTeleportStarter extends BridgeTransferStarter {
+  public transferType: TransferType = 'eth_teleport'
+
+  constructor(props: BridgeTransferStarterProps) {
+    super(props)
+  }
+
+  public async requiresNativeCurrencyApproval({
+    amount,
+    signer
+  }: RequiresNativeCurrencyApprovalProps) {
+    return false
+  }
+
+  public async approveNativeCurrency({
+    signer,
+    amount
+  }: ApproveNativeCurrencyProps) {
+    // const ethBridger = await EthBridger.fromProvider(
+    //   this.destinationChainProvider
+    // )
+    // return ethBridger.approveGasToken({
+    //   l1Signer: signer,
+    //   amount
+    // })
+  }
+
+  public async approveNativeCurrencyEstimateGas() {
+    // no-op
+  }
+
+  public requiresTokenApproval = async () => false
+
+  public approveTokenEstimateGas = async () => {
+    // no-op
+  }
+
+  public approveToken = async () => {
+    // no-op
+  }
+
+  public async transferEstimateGas({ amount, signer }: TransferEstimateGas) {
+    // get the intermediate L2 chain provider
+    const { l2Provider } = await getL2ConfigForTeleport({
+      destinationChainProvider: this.destinationChainProvider
+    })
+
+    const l3Network = await getL2Network(this.destinationChainProvider)
+    const l1l3Bridger = new EthL1L3Bridger(l3Network)
+
+    const depositRequest = await l1l3Bridger.getDepositRequest({
+      l1Signer: signer,
+      amount: amount,
+      l1Provider: this.sourceChainProvider,
+      l2Provider,
+      l3Provider: this.destinationChainProvider
+    })
+
+    const estimatedParentChainGas = await this.sourceChainProvider.estimateGas(
+      depositRequest.txRequest
+    )
+
+    return {
+      estimatedParentChainGas,
+      estimatedChildChainGas: constants.Zero
+    }
+  }
+
+  public async transfer({ amount, signer, destinationAddress }: TransferProps) {
+    const address = await getAddressFromSigner(signer)
+
+    const l3Network = await getL2Network(this.destinationChainProvider)
+
+    // get the intermediate L2 chain provider
+    const { l2Provider } = await getL2ConfigForTeleport({
+      destinationChainProvider: this.destinationChainProvider
+    })
+
+    const l1l3Bridger = new EthL1L3Bridger(l3Network)
+
+    const depositRequest = await l1l3Bridger.getDepositRequest({
+      l1Signer: signer,
+      to: destinationAddress ?? address,
+      amount: amount,
+      l1Provider: this.sourceChainProvider,
+      l2Provider,
+      l3Provider: this.destinationChainProvider
+      // l2TicketGasOverrides: {
```

Comment:
To test the redemption flows, uncomment these lines and the teleporter tx will fail and require redemption.

Reply:
There is a similar (not exactly the same code) process for ERC20-Teleporter flow